controllers/components/kserve/kserve_support.go

```diff
@@ -140,6 +158,41 @@ func defineServiceMeshFeatures(ctx context.Context, cli client.Client, dscispec
 	}
 }
 
+func getTemplateData(ctx context.Context, cli client.Client, k *componentApi.Kserve, dsci *dsciv1.DSCInitialization) map[string]any {
+	knativeIngressDomain := getKnativeDomain(ctx, cli, k)
+	knativeCertificateSecret := getKnativeCertSecretName(k)
+
+	return map[string]any{
```

Comment:
nit: the component and dsci are already [part of the template data](https://github.com/opendatahub-io/opendatahub-operator/blob/72443c53d3210b80a56ae25669644a75a5eb2bce/pkg/controller/actions/render/template/action_render_templates.go#L103-L105), so it may not be necessary to add additional elements (but would require to change the templates)

Reply:
Yeah, I noticed that, and had originally started down that path of modifying the templates here also. I hit some issues that I didn't care to troubleshoot at the time, so I instead just declared the data that the current templates expect, for now. I could try to address that here, or alternatively it could be done in a follow-up PR so as not to bloat this one (I slightly prefer the idea of that being in a separate PR, because of how we squash commits on merge here...I think it's a reasonable thing to have as a separate commit).

Reply:
FWIW, there is now a `gc.WithDataFn` for dynamic content